src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -63,6 +66,21 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
             return actionArguments;
         }
 
+        // Called via reflection.
+        private static void CallPropertyAddRange<TElement>(object target, object source)
+        {
+            var targetCollection = (ICollection<TElement>)target;
+            var sourceCollection = source as IEnumerable<TElement>;
+            if (sourceCollection != null && !targetCollection.IsReadOnly)
+            {
+                targetCollection.Clear();
+                foreach (var item in sourceCollection)
+                {
+                    targetCollection.Add(item);
+                }
+            }
+        }
```

Comment:
> understand if you want to leave this code as-is for consistency with the legacy

yeah, let's leave refactoring the core model binding classes until after we have lots more tests in place